LEAF_Request_Portal/sources/FormWorkflow.php

```diff
@@ -218,12 +242,13 @@ public function getActionable(object $form, array $records): array {
     }
 
     /**
-     * Supports getCurrentSteps()
+     * getRecordsDependencyData retrieves dependency data related to records
      * @param object $form instance of Form
      * @param array $records result set from a query on db:records. Requires 'recordID'.
+     * @param bool optional Only select unfilled dependencies
      * @return array $records list of dependencies * records
      */
-    private function retrieveRecordDependencies(object $form, array $records): array {
+    public function getRecordsDependencyData(object $form, array $records, bool $selectUnfilled = false): array {
         $numRecords = count($records);
         if ($numRecords == 0) {
             return $records;

```

Comment:
Would like this method to get modernized a bit more.  There should be only one return.  And we're moving to a json/array format where there are at least two parts.  1 telling the success or fail with data and 1 with the returned data.  How much effort would it be to update this method and its callers?

Reply:
I think we should formalize a standard response class first, as part of a separate modernization task. Right now the response format is being implemented ad-hoc and can potentially be inconsistent.